src/router/set.js

```diff
@@ -13,131 +13,144 @@ var normalizePathSets = require('../operations/ranges/normalizePathSets');
-                //     incoming jsonGraph.
-                // 3.  test intersection against incoming optimized path and
-                //     unhandledPathSet
-                // 4.  If 3 is true, build the jsonGraphFragment by using a
-                //     pathValue of optimizedPath and vale from un-optimized
-                //     path and original jsonGraphEnvelope.
-                var jsonGraphEnvelope = {jsonGraph: jsonGraphFragment};
-                var unhandledPathsTree = unhandledPaths.
-                    reduce(function(acc, path) {
-                        pathValueMerge(acc, {path: path, value: null});
-                        return acc;
-                    }, {});
-
-                // 1. Spread
-                var pathIntersection = spreadPaths(jsonGraph.paths).
-
-                    // 2.1 Optimize.  We know its one at a time therefore we
-                    // just pluck [0] out.
-                    map(function(path) {
-                        return [
-                            // full path
-                            path,
-
-                            // optimized path
-                            optimizePathSets(details.jsonGraph, [path],
-                                                router.maxRefFollow)[0]]
-                    }).
-
-                    // 2.2 Remove all the optimized paths that were found in
-                    // the cache.
-                    filter(function(x) { return x[1]; }).
-
-                    // 3.1 test intersection.
-                    map(function(pathAndOPath) {
-                        var oPath = pathAndOPath[1];
-                        var hasIntersection = hasIntersectionWithTree(
-                            oPath, unhandledPathsTree);
-
-                        // Creates the pathValue if there are a path
-                        // intersection
-                        if (hasIntersection) {
-                            var value =
-                                getValue(jsonGraph.jsonGraph, pathAndOPath[0]);
-
-                            return {
-                                path: oPath,
-                                value: value
-                            };
-                        }
-
-                        return null;
-                    }).
-
-                    // 3.2 strip out nulls (the non-intersection paths).
-                    filter(function(x) { return x !== null; });
-
-                    // 4. build the optimized JSONGraph envelope.
-                    pathIntersection.
-                        reduce(function(acc, pathValue) {
-                            pathValueMerge(acc, pathValue);
+
+    return Observable.defer(function() {
+        var jsongCache = {};
+        var action = runSetAction(router, jsonGraph, jsongCache);
+        jsonGraph.paths = normalizePathSets(jsonGraph.paths);
+
+        if (getPathsCount(jsonGraph.paths) > router.maxPaths) {
```

Comment:
Same here? I imagine this is the exact same set impl as before just with the path check, on subscription?